testify/test_case.py

```diff
@@ -337,6 +338,32 @@ def __run_test_methods(self, class_fixture_failures):
                     if self.failure_limit and self.failure_count >= self.failure_limit:
                         break
 
+    def addfinalizer(self, teardown_func):
+        if self._stage in (self.STAGE_SETUP, self.STAGE_TEST_METHOD, self.STAGE_TEARDOWN):
+            self.__extra_test_teardowns.append(teardown_func)
+        elif self._stage in (self.STAGE_CLASS_SETUP, self.STAGE_CLASS_TEARDOWN):
+            self.__extra_class_teardowns.append(teardown_func)
+        else:
+            raise RuntimeError('Tried to add a teardown while the test was not being executed.')
+
+    @test_fixtures.class_setup
+    def __setup_extra_class_teardowns(self):
+        self.__extra_class_teardowns = []
+
+    @test_fixtures.setup
+    def __setup_extra_test_teardowns(self):
+        self.__extra_test_teardowns = []
+
+    @test_fixtures.teardown
+    def __teardown_extra_test_teardowns(self):
+        for teardown in self.__extra_test_teardowns:
+            teardown()
+
+    @test_fixtures.class_teardown
+    def __teardown_extra_class_teardowns(self):
+        for teardown in self.__extra_class_teardowns:
+            teardown()
+
```

Comment:
hrmmm there's certainly benefits to reusing the setup/teardown code but I think these actually belong inside the infrastructure instead of on this class.  @bukzor what do you think?